pkg/controller/logcollector/logcollector_controller.go

```diff
@@ -301,9 +367,23 @@ func (r *ReconcileLogCollector) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{RequeueAfter: 30 * time.Second}, nil
 	}
 
-	// Everything is available - update the CRD status.
+	// Update the LogCollector instance with any changes that have occurred.
+	if isModified {
```

Comment:
This update should actually come before we render and apply resources, immediately after any defaulting is done.

Reply:
If rendering fails, the user should be able to query the CR to see exactly what state was being used to render.

Reply:
Right. I'll move it above the Fleuntd related rendering.